File: expconslayout/src/main/java/rjsv/expconslayout/ExpandableConstraintLayoutListener.java
package rjsv.expconslayout;

import rjsv.expconslayout.enumerators.ExpandableConstraintLayoutStatus;

/**
 * @author <a href="mailto:rvfts@hotmail.com"/>
 * @version $Revision : 1 $
 */

public interface ExpandableConstraintLayoutListener {
    /**
     * Notifies the start of the animation.
     * Sync from android.animation.Animator.AnimatorListener.onAnimationStart(Animator animation)
     */
    void onAnimationStart(ExpandableConstraintLayoutStatus status);

    /**
     * Notifies the end of the animation.
     * Sync from android.animation.Animator.AnimatorListener.onAnimationEnd(Animator animation)
     */
    void onAnimationEnd(ExpandableConstraintLayoutStatus status);

    /**
     * Notifies the layout is going to open.
     */
    void onPreOpen();

    /**
     * Notifies the layout is going to equal close size.
     */
    void onPreClose();

    /**
     * Notifies the layout opened.
     */
    void onOpened();

    /**
     * Notifies the layout size equal closed size.
     */
    void onClosed();

}

File: expconslayout/src/main/java/rjsv/expconslayout/ExpandableConstraintLayoutListener.kt
package rjsv.expconslayout

import rjsv.expconslayout.enumerators.ExpandableConstraintLayoutStatus

/**
 * @author [](mailto:rvfts@hotmail.com)
 * @version $Revision : 1 $
 */

interface ExpandableConstraintLayoutListener {
    /**
     * Notifies the start of the animation.
     * Sync from android.animation.Animator.AnimatorListener.onAnimationStart(Animator animation)
     */
    fun onAnimationStart(status: ExpandableConstraintLayoutStatus)

    /**
     * Notifies the end of the animation.
     * Sync from android.animation.Animator.AnimatorListener.onAnimationEnd(Animator animation)
     */
    fun onAnimationEnd(status: ExpandableConstraintLayoutStatus)

    /**
     * Notifies the layout is going to open.
     */
    fun onPreOpen()

    /**
     * Notifies the layout is going to equal close size.
     */
    fun onPreClose()

    /**
     * Notifies the layout opened.
     */
    fun onOpened()

    /**
     * Notifies the layout size equal closed size.
     */
    fun onClosed()

}